packages/interface/src/components/action-item/index.js

```diff
@@ -1,27 +1,37 @@
 /**
  * External dependencies
  */
-import { isEmpty, noop } from 'lodash';
+import { isArray, isEmpty, noop } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { ButtonGroup, Button, Slot, Fill } from '@wordpress/components';
+import deprecated from '@wordpress/deprecated';
 import { Children } from '@wordpress/element';
 
 function ActionItemSlot( {
 	name,
-	as = [ ButtonGroup, Button ],
+	as: Component = ButtonGroup,
 	fillProps = {},
 	bubblesVirtually,
 	...props
 } ) {
-	const [ Container, Item ] = as;
+	if ( isArray( Component ) ) {
+		deprecated(
+			'Passing a tuple of components with `as` prop to `ActionItem.Slot` component',
+			{
+				alternative: 'a component with `as` prop',
+				version: '10.3',
+			}
+		);
+		Component = Component[ 0 ];
```

Comment:
If an `array` was passed, shouldn't we get `Component[ 1 ]` to set `as` in `Slot` below? 🤔 

First time I'm exploring this part of the codebase, so still trying to understand :)

Reply:
It this part of the code:

<img width="331" alt="Screen Shot 2021-02-26 at 09 34 49" src="https://user-images.githubusercontent.com/699132/109276129-ecf41e80-7815-11eb-8d40-e31687f2f26a.png">

so it translates to `Component[ 0 ]`. It tests well for me.